posthog/utils.py

```diff
@@ -871,9 +872,18 @@ def get_milliseconds_between_dates(d1: dt.datetime, d2: dt.datetime) -> int:
     return abs(int((d1 - d2).total_seconds() * 1000))
 
 
-def encode_get_request_params(data: Dict[str, Any]) -> Dict[str, str]:
+def encode_value_as_param(value: Union[str, list, dict]) -> str:
+    if isinstance(value, (list, dict, tuple)):
+        return json.dumps(value, cls=DataclassJSONEncoder)
+    elif isinstance(value, Enum):
+        return value.value
+    else:
+        return value
+
+
+def encode_get_request_params(data: Dict[str, Any], func: Callable = encode_value_as_param) -> Dict[str, str]:
```

Comment:
I'm not quite sure what the issue is, but I would expect encoding to always be the same for the same type of input. Is it possible that we have two types of input for `data` but we're treating them the same?

It it possible to clarify in the test a little more? I seems we already check for `value is not None`, but is this something like `value == "None"` instead of `value == None`?

Reply:
yup exactly. Prior to using encode_get_request_params, our actor URLs were formatting in entity_math=None and later `entity_math` would be processed as a string instead of a boolean

Reply:
Also, is it worth splitting out this alternative behaviour as a separate function, like `encode_get_request_params_in_this_specific_case`?

Reply:
Do you know at which point `entity_math` is converted to a string? I feel like `Filter` handling should have validated data here and provided a predictable structure 🤔  And `"None"` should not have been valid in the first place?

Reply:
It's happening right [here](https://github.com/PostHog/posthog/pull/8103/files#diff-39fc0e4caa1edce343b310996c2083bb45e2da87e6c31472c7924049c1f6eeaeR96) right before we parse it into a url

Reply:
Looks like the typing on `entity.math` is `Optional[MATH_TYPE]`? But somehow it's ending up as `"None"`?

Reply:
omg I totally missed that we went calling `encode_get_request_params` at all before. :dense: 

It's still not obvious what the lambda addition is all about